runtime/lua/vim/shared.lua

```diff
@@ -44,9 +44,9 @@ end
 --@param plain If `true` use `sep` literally (passed to String.find)
 --@returns Iterator over the split components
 local function gsplit(s, sep, plain)
-  assert(type(s) == "string")
-  assert(type(sep) == "string")
-  assert(type(plain) == "boolean" or type(plain) == "nil")
+  assert(type(s) == "string", string.format("String type argument expected, got '%s'", type(s)))
+  assert(type(sep) == "string", string.format("String type argument expected, got '%s'", type(sep)))
+  assert(type(plain) == "boolean" or type(plain) == "nil", string.format("Boolean or nil type argument expected, got '%s'", type(plain)))
```

Comment:
We can make these messages less verbose without losing clarity.

```
Expected string,, got '%s'
Expected boolean or nil,, got '%s'
```

same for the other cases.